app/src/main/java/com/kickstarter/libs/utils/RewardUtils.kt

```diff
@@ -0,0 +1,209 @@
+     */
+    fun isLimitReached(reward: Reward): Boolean {
+        val remaining = reward.remaining()
+        return reward.limit() != null && remaining != null && remaining <= 0
+    }
+
+    /**
+     * Returns `true` if the reward is considered the 'non-reward' option, i.e. the reward is the option
+     * backers select when they want to pledge to a project without selecting a particular reward.
+     */
+    fun isNoReward(reward: Reward): Boolean {
+        return reward.id() == 0L
+    }
+
+    /**
+     * Returns `true` if the reward is a specific reward for a project, i.e. it is not the 'no-reward' option.
+     */
+    fun isReward(reward: Reward): Boolean {
+        return !isNoReward(reward)
+    }
+
+    /**
+     * Returns `true` if the reward has shipping enabled, `false` otherwise.
+     */
+    fun isShippable(reward: Reward): Boolean {
+        val shippingType = reward.shippingType()
+        val noShippingTypes = reward.shippingPreferenceType() == Reward.ShippingPreference.NONE
+        return shippingType != null && !(Reward.SHIPPING_TYPE_NO_SHIPPING == shippingType || noShippingTypes)
+    }
+
+    /**
+     * Returns `true` if the reward is a Digital Reward, meaning tha it has "shippingPreference": "none"
+     * if the model is a response from GraphQL or "shippingPreference": "no_shipping" if the model is
+     * a response from V1
+     * @param reward
+     * @return isDigital: true or false
+     */
+    fun isDigital(reward: Reward): Boolean {
+        val isDigitalV1 = reward.shippingType() != null && reward.shippingType().equals(Reward.SHIPPING_TYPE_NO_SHIPPING, ignoreCase = true)
+        return (reward.shippingPreferenceType() == Reward.ShippingPreference.NONE || reward.shippingPreferenceType() == Reward.ShippingPreference.NOSHIPPING ||
+                isDigitalV1) && !isShippable(reward)
+    }
+
+    /**
+     * Returns `true` if the reward has a valid expiration date.
+     */
+    fun isTimeLimited(reward: Reward): Boolean {
+//         TODO: 2019-06-14 remove epoch check after Garrow fixes `current` bug in backend
+        return reward.endsAt() != null && !DateTimeUtils.isEpoch(reward.endsAt()!!)
+    }
+
+    /**
+     * Returns unit of time remaining in a readable string, e.g. `days to go`, `hours to go`.
+     */
+    fun deadlineCountdownDetail(reward: Reward, context: Context,
+                                ksString: KSString): String {
+        return ksString.format(context.getString(R.string.discovery_baseball_card_time_left_to_go),
+                "time_left", deadlineCountdownUnit(reward, context)
+        )
+    }
+
+    /**
+     * Returns the most appropriate unit for the time remaining until the reward
+     * reaches its deadline.
+     *
+     * @param context an Android context.
+     * @return the String unit.
+     */
+    fun deadlineCountdownUnit(reward: Reward, context: Context): String {
+        val seconds = timeInSecondsUntilDeadline(reward)
+        return when {
```

Comment:
Great! the 'when' expression is hard to use great job here :)

Reply:
Thank you!